Providers/Profile.swift

```diff
@@ -964,9 +964,12 @@ open class BrowserProfile: Profile {
         }
 
         fileprivate func syncClientsWithDelegate(_ delegate: SyncDelegate, prefs: Prefs, ready: Ready, why: SyncReason) -> SyncResult {
-            log.debug("Syncing clients to storage.")
-            let clientSynchronizer = ready.synchronizer(ClientsSynchronizer.self, delegate: delegate, prefs: prefs, why: why)
-            return clientSynchronizer.synchronizeLocalClients(self.profile.remoteClientsAndTabs, withServer: ready.client, info: ready.info)
+            log.debug("Updating the FxA devices list.")
+            return self.profile.account!.updateFxADevices(withDb: self.profile.db).bind { _ in
```

Comment:
Do you really want to fetch and update the FxA device list every time we even consider syncing clients? Remember that `synchronizeLocalClients` won't do any work unless we need to, and your `updateFxADevices` flow here not only hits the network, but also writes to the DB. That's power-hungry.

Consider reworking this: can you update the device list only if the clients sync actually did work (see the `SyncResult`)? How does FxA do rate-limiting and backoff?